src/serialisation/wire_msg_header.rs

```diff
@@ -26,45 +29,93 @@ pub(crate) struct WireMsgHeader {
-const HDR_VERSION_BYTES_START: usize = 2;
-const HDR_VERSION_BYTES_END: usize = 4;
+const HDR_VERSION_BYTES_START: usize = HDR_SIZE_BYTES_LEN;
 const HDR_VERSION_BYTES_LEN: usize = size_of::<u16>();
+const HDR_VERSION_BYTES_END: usize = HDR_VERSION_BYTES_START + HDR_VERSION_BYTES_LEN;
 
 // Bytes index in the header for the 'kind' field
-const HDR_KIND_BYTES_START: usize = 4;
+const HDR_KIND_BYTES_START: usize = HDR_VERSION_BYTES_END;
+const HDR_KIND_BYTES_LEN: usize = 1;
+
+// Bytes index in the header for the 'dest' field
+const HDR_DEST_BYTES_START: usize = HDR_KIND_BYTES_START + HDR_KIND_BYTES_LEN;
+const HDR_DEST_BYTES_LEN: usize = XOR_NAME_LEN;
+const HDR_DEST_BYTES_END: usize = HDR_DEST_BYTES_START + HDR_DEST_BYTES_LEN;
+
+// Bytes index in the header for the 'dest_section_pk' field
+const HDR_DEST_PK_BYTES_START: usize = HDR_DEST_BYTES_END;
+const HDR_DEST_PK_BYTES_LEN: usize = PK_SIZE;
+const HDR_DEST_PK_BYTES_END: usize = HDR_DEST_PK_BYTES_START + HDR_DEST_PK_BYTES_LEN;
+
+// Bytes index in the header for the 'src_section_pk' field
+const HDR_SRC_PK_BYTES_START: usize = HDR_DEST_PK_BYTES_END;
+const HDR_SRC_PK_BYTES_LEN: usize = PK_SIZE;
+const HDR_SRC_PK_BYTES_END: usize = HDR_SRC_PK_BYTES_START + HDR_SRC_PK_BYTES_LEN;
+
+const HEADER_MIN_SIZE: usize = HDR_SIZE_BYTES_LEN
+    + HDR_VERSION_BYTES_LEN
+    + HDR_KIND_BYTES_LEN
+    + HDR_DEST_BYTES_LEN
+    + HDR_DEST_PK_BYTES_LEN;
 
 impl WireMsgHeader {
     // Instantiate a WireMsgHeader as per current supported version.
-    pub fn new(kind: MessageKind) -> Self {
+    pub fn new(
+        kind: MessageKind,
+        dest: XorName,
+        dest_section_pk: PublicKey,
+        src_section_pk: Option<PublicKey>,
+    ) -> Self {
         Self {
-            header_size: Self::size() as u16,
+            header_size: Self::bytes_size(src_section_pk.is_some()) as u16,
             version: MESSAGING_PROTO_VERSION,
             kind,
+            dest,
+            dest_section_pk,
+            src_section_pk,
         }
     }
 
+    // Return the kind of this message
     pub fn kind(&self) -> MessageKind {
         self.kind
     }
 
+    // Return the destination section PublicKey for this message
+    pub fn dest_section_pk(&self) -> PublicKey {
+        self.dest_section_pk
+    }
+
+    // Return the destination for this message
+    pub fn dest(&self) -> XorName {
+        self.dest
+    }
+
+    // Return the source section PublicKey for this message
+    // if it's a NodeMsg and it was included in the header
+    pub fn src_section_pk(&self) -> Option<PublicKey> {
+        self.src_section_pk
+    }
+
     // Parses the provided bytes to deserialize a WireMsgHeader,
     // returning the created WireMsgHeader, as well as the remaining bytes which
     // correspond to the message payload. The caller shall then take care of
     // deserializing the payload using the information provided in the WireMsgHeader.
     pub fn from(mut bytes: Bytes) -> Result<(Self, Bytes)> {
```

Comment:
I think it's best to keep this function definition consistent with the `From` trait and rename this.